### ISSUE TITLE
[0.5.0] Check handler do not execute

### ISSUE THREAD

**Issue by HeapUnderfl0w** [zeyla/serenity#243]
Serenity Version 0.5.0 (git branch) on commit 26fe139363a847542bbe609fe4d15accbf4fef14

I have the current problem:
I am unable to do write anything to stdout while executing in the check handler of the Command.

The command is registered like this (the variable `g` has the type `CreateGroup`)
```rust
g = g.command("user", |f| {
	f.exec(command::inspect::user)
	.bucket("5s")
	.desc(
		"Get information about the current or another user.\n\
		 **Params**: [userid | username | username#discriminator]",
	)
	.check(|_, msg, _, _| {

		let stdout = io::stdout();
		let mut handle = stdout.lock();

		let _ = handle.write(b"Hello from the Listener !");

		if ap(
			has_feature_override("com_user", &msg.author.id.0),
			PType::Error,
		) {
			return true;
		}
		ap(owner_only_helper(&msg.author.id.0), PType::Error)
	})
});
```
I also tried using
```rust
println!("Hello from the Listener !");
```
and
```rust
// Using the log & log4rs crates
error!("Hello from the Listener !");
```
to print something to stdout but nothing appears on the console.

**Comment by HeapUnderfl0w** [comment]
TL;DR I assume the check handler's are never executed.

I have tested the code-paths accessible to me by placing panics in them (the .check handler & the actual command function).

I was able to confirm that there was no general persistent lock of stdout as both the `on_message` and the actual command handler could print to stdout
![Image of log](https://vgy.me/RhU3DS.png)

Then i tried to get any message out using File IO. I did not even see the creation of a file.

After that i got the suspicion that the check handler was never executed.
I used 2 tests to confirm this.

In the first test i simply put a `return false` on the first line of the handler.  
The command did still execute.

In the second test i put a `panic!();` in the first line of both the check handler & the actual command function.
```
thread 'serenity client' panicked at 'Explicit panic in COMMAND handler of USER', src\command\inspect\user.rs:18:5
```
As seen in the error message above, "only" the actual command function did panic.

TODO to self: Test on other handlers.

**Comment by HeapUnderfl0w** [comment]
This has been resolved in a recent push of the git repo.
